presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java

```diff
@@ -153,9 +153,9 @@ else if (outputRequired || !fieldsWithFilters.isEmpty()) {
                 StreamDescriptor nestedStream = nestedStreams.get(i);
                 String fieldName = nestedStream.getFieldName().toLowerCase(Locale.ENGLISH);
                 Optional<Type> fieldOutputType = nestedTypes.isPresent() ? Optional.of(nestedTypes.get().get(i)) : Optional.empty();
-                boolean requiredField = requiredFields.map(names -> names.containsKey(fieldName)).orElse(true);
+                boolean requiredField = requiredFields.map(names -> names.containsKey(fieldName)).orElse(outputRequired || fieldsWithFilters.contains(fieldName));
```

Comment:
nit: it might be easier to read if the change was only to replace `true` with `outputRequired` and `fieldsWithFilters.contains(fieldName)` check is left in its original place